### PR TITLE
skip k8s insight collection if remoteconfig is disabled

### PR DESCRIPTION
**Pull request by huseyinbabal**
## Description

Changes proposed in this pull request:

- k8s insight collection is started even with OSS installation. Added condition to not start

## Testing

1. Run helm botkube with debug log enabled
2. You shouldn't see `time="2023-03-31T11:11:15Z" level=debug msg="Collecting Kubernetes insights"`

### CODE REVIEW

File: cmd/botkube/main.go [hunk 1]
```diff
@@ -388,7 +388,7 @@ func run(ctx context.Context) error {
 		}()
 		heartbeatReporter := heartbeat.GetReporter(remoteCfgEnabled, logger, gqlClient)
 		k8sCollector := insights.NewK8sCollector(k8sCli, heartbeatReporter, logger, reportHeartbeatInterval, reportHeartbeatMaxRetries)
-		return k8sCollector.Start(ctx)
```
Comment (mszostok) [inline cmd/botkube/main.go:391]: why not to simply do the 
```go
	if !remoteCfgEnabled {
		k.logger.Debug("Remote config is not enabled, skipping k8s insights collection...")
		return nil
	}
```

at the beginning of the func? As a result also the `heartbeatReporter` and `k8sCollector` won't be created at all. Also the  `heartbeat.GetReporter` could be simplified.

Comment (huseyinbabal) [reply cmd/botkube/main.go:391]: How can't I see that :) on it...

Comment (huseyinbabal) [reply cmd/botkube/main.go]: Updated, please check

Comment (huseyinbabal) [reply cmd/botkube/main.go:391]: @mszostok as you suggested, I also simplified reporter

**Review by mszostok — APPROVED** [review]

**Pull request merged**